subsys/net/lib/lwm2m_client_utils/lwm2m/location_assistance_obj.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_obj_loc_assistance
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <string.h>
+#include <init.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include <net/lwm2m_client_utils.h>
+
+#define LOCATION_ASSIST_VERSION_MAJOR 1
+#define LOCATION_ASSIST_VERSION_MINOR 0
+
+/* Location Assistance resource IDs */
+#define LOCATION_ASSIST_ASSIST_TYPE			0
+#define LOCATION_ASSIST_AGPS_MASK			1
+#define LOCATION_ASSIST_PGPS_PRED_COUNT			2
+#define LOCATION_ASSIST_PGPS_PRED_INTERVAL		3
+#define LOCATION_ASSIST_PGPS_START_GPS_DAY		4
+#define LOCATION_ASSIST_PGPS_START_GPS_TIME_OF_DAY	5
+#define LOCATION_ASSIST_ASSIST_DATA			6
+#define LOCATION_ASSIST_RESULT_CODE			7
+#define LOCATION_ASSIST_LATITUDE			8
+#define LOCATION_ASSIST_LONGITUDE			9
+#define LOCATION_ASSIST_ALTITUDE			10
+#define LOCATION_ASSIST_ACCURACY			11
+
+#define LOCATION_ASSIST_MAX_ID				12
+
+static int32_t assist_type;
+static int32_t agps_mask;
+static int32_t pgps_pred_count;
+static int32_t pgps_pred_interval;
+static int32_t pgps_start_gps_day;
+static int32_t pgps_start_gps_time_of_day;
+//Assistance data
+static char assist_data[1024];
+static int32_t result;
+static float latitude;
+static float longitude;
+static float altitude;
+static float accuracy;
```

Comment:
We should use `double` for those - a bit counter-intuitive but engine represents LwM2M float type as double for best accuracy.